src/gui/properties/speedplotview.cpp

```diff
@@ -71,62 +75,94 @@ SpeedPlotView::SpeedPlotView(QWidget *parent)
 void SpeedPlotView::setGraphEnable(GraphID id, bool enable)
 {
     m_properties[id].m_enable = enable;
+    this->viewport()->update();
 }
 
-void SpeedPlotView::pushXPoint(double x)
+void SpeedPlotView::pushPoint(SpeedPlotView::PointData point)
 {
-    while (m_xData.size() >= m_maxCapacity)
-        m_xData.pop_front();
+    m_counter30Min = (m_counter30Min + 1) % 3;
+    m_counter6Hour = (m_counter6Hour + 1) % 6;
 
-    m_xData.append(x);
-}
+    m_data5Min.push_back(point);
 
-void SpeedPlotView::pushYPoint(GraphID id, double y)
-{
-    while (m_yData[id].size() >= m_maxCapacity)
-        m_yData[id].pop_front();
+    if (m_counter30Min == 0) {
+        m_data30Min.push_back(point);
+    }
+    else {
+        m_data30Min.back().x = (m_data30Min.back().x * m_counter30Min + point.x) / (m_counter30Min + 1);
+        for (int id = UP; id < NB_GRAPHS; ++id) {
+            m_data30Min.back().y[id] = (m_data30Min.back().y[id] * m_counter30Min + point.y[id]) / (m_counter30Min + 1);
+        }
+    }
 
-    m_yData[id].append(y);
+    if (m_counter6Hour == 0) {
+        m_data6Hour.push_back(point);
+    }
+    else {
+        m_data6Hour.back().x = (m_data6Hour.back().x * m_counter6Hour + point.x) / (m_counter6Hour + 1);
+        for (int id = UP; id < NB_GRAPHS; ++id) {
+            m_data6Hour.back().y[id] = (m_data6Hour.back().y[id] * m_counter6Hour + point.y[id]) / (m_counter6Hour + 1);
+        }
+    }
 }
 
 void SpeedPlotView::setViewableLastPoints(TimePeriod period)
 {
+    m_period = period;
+
     switch (period) {
     case SpeedPlotView::MIN1:
-        m_viewablePointsCount = SpeedPlotView::MIN1_SEC;
+        m_viewablePointsCount = MIN1_SEC;
         break;
     case SpeedPlotView::MIN5:
-        m_viewablePointsCount = SpeedPlotView::MIN5_SEC;
+        m_viewablePointsCount = MIN5_SEC;
         break;
     case SpeedPlotView::MIN30:
-        m_viewablePointsCount = SpeedPlotView::MIN30_SEC;
+        m_viewablePointsCount = MIN30_BUF_SIZE;
         break;
     case SpeedPlotView::HOUR6:
-        m_viewablePointsCount = SpeedPlotView::HOUR6_SEC;
-        break;
-    default:
+        m_viewablePointsCount = HOUR6_BUF_SIZE;
         break;
     }
+
+    this->viewport()->update();
 }
 
 void SpeedPlotView::replot()
 {
-    this->viewport()->update();
+    if (m_period == MIN1 || m_period == MIN5 ||
+            (m_period == MIN30 && m_counter30Min == 2) ||
+            (m_period == HOUR6 && m_counter6Hour == 5))
+        this->viewport()->update();
 }
 
-double SpeedPlotView::maxYValue()
+boost::circular_buffer<SpeedPlotView::PointData> &SpeedPlotView::getCurrentData()
 {
-    double maxYValue = 0;
-    for (QMap<GraphID, QQueue<double> >::const_iterator it = m_yData.begin(); it != m_yData.end(); ++it) {
+    switch (m_period) {
+    default:
```

Comment:
@onto
In case of coding style (not sure it covers this case), just separate `default:` from other `case:` and move it to bottom of the switch block.
ps. no problem about program validness, just style problem.